bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java

```diff
@@ -0,0 +1,352 @@
+                getDevices();
+            }
+        } catch (InterruptedException e) {
+        }
+    }
+
+    private void login() throws InterruptedException {
+        HttpResult result = sendRequest(BASE_URL + "/v5/Login", HttpMethod.POST, null,
+                new StringContentProvider(gsonUpperCase.toJson(new LoginRequestDTO(username, password))),
+                "application/json");
+        if (HttpStatus.isSuccess(result.responseCode)) {
+            LoginResponseDTO loginResponse = gsonUpperCase.fromJson(result.content, LoginResponseDTO.class);
+            if (loginResponse != null) {
+                securityToken = loginResponse.securityToken;
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Invalid Response Body");
+            }
+        } else {
+            handleErrorResponse(result);
+            if (thing.getStatusInfo().getStatusDetail() == ThingStatusDetail.CONFIGURATION_ERROR) {
+                // bad credentials, stop trying to login
+                stopPolls();
+            }
+        }
+    }
+
+    private void getAccount() throws InterruptedException {
+        HttpResult result = sendRequest(BASE_URL + "/v5/My?expand=account", HttpMethod.GET, securityToken, null, null);
+        if (HttpStatus.isSuccess(result.responseCode)) {
+            account = gsonUpperCase.fromJson(result.content, AccountDTO.class);
+        } else {
+            handleErrorResponse(result);
+        }
+    }
+
+    private void getDevices() throws InterruptedException {
+        AccountDTO localAccount = account;
+        if (localAccount == null) {
+            return;
+        }
+        HttpResult result = sendRequest(String.format("%s/v5.1/Accounts/%s/Devices", BASE_URL, localAccount.account.id),
+                HttpMethod.GET, securityToken, null, null);
+        if (HttpStatus.isSuccess(result.responseCode)) {
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+            DevicesDTO devices = gsonLowerCase.fromJson(result.content, DevicesDTO.class);
+            if (devices != null) {
+                devicesCache = devices;
+                devices.items.forEach(device -> {
+                    ThingTypeUID thingTypeUID = new ThingTypeUID(BINDING_ID, device.deviceFamily);
+                    if (SUPPORTED_DISCOVERY_THING_TYPES_UIDS.contains(thingTypeUID)) {
+                        for (Thing thing : getThing().getThings()) {
+                            ThingHandler handler = thing.getHandler();
+                            if (handler != null && ((MyQDeviceHandler) handler).getSerialNumber()
+                                    .equalsIgnoreCase(device.serialNumber)) {
+                                ((MyQDeviceHandler) handler).handleDeviceUpdate(device);
+                            }
+                        }
```

Comment:
This seems the same code as in line 126. Can you make it re-useable?

Reply:
it looks similar, but different enough that i prefer to keep it the way it is